security/pkg/k8s/chiron/utils.go

```diff
@@ -89,15 +92,30 @@ func GenKeyCertK8sCA(certClient certclient.CertificateSigningRequestInterface, d
 		return nil, nil, nil, err
 	}
 	csrName := getRandomCsrName(secretName, secretNamespace)
-	csrSpec := &cert.CertificateSigningRequestSpec{
-		Request: csrPEM,
-		Groups:  []string{"system:authenticated"},
-		Usages: []cert.KeyUsage{
-			cert.UsageDigitalSignature,
-			cert.UsageKeyEncipherment,
-			cert.UsageServerAuth,
-			cert.UsageClientAuth,
-		},
+
+	if k8sSigner == globalconstants.DefaultK8SSigner {
+		csrSpec = &cert.CertificateSigningRequestSpec{
+			Request: csrPEM,
+			Groups:  []string{"system:authenticated"},
+			Usages: []cert.KeyUsage{
+				cert.UsageDigitalSignature,
+				cert.UsageKeyEncipherment,
+				cert.UsageServerAuth,
+				cert.UsageClientAuth,
+			},
+		}
+	} else {
+		csrSpec = &cert.CertificateSigningRequestSpec{
+			SignerName: k8sSigner,
```

Comment:
As the PR adds the support of new types of signers, consider adding integration tests for each supported signer to verify a new signer is working as expected.

Reply:
There is no in-built K8s signer that can provide this functionality currently  (apart from legacy-unknown that is being deprecated in K8s 1.22). This is meant to serve as a placeholder for future signers/external CA's that can provide this functionality